centreon/src/Core/Host/Application/UseCase/DeleteHost/DeleteHost.php

```diff
@@ -0,0 +1,149 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Host\Application\UseCase\DeleteHost;
+
+use Centreon\Domain\Contact\Contact;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Domain\Repository\Interfaces\DataStorageEngineInterface;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Application\Common\UseCase\NoContentResponse;
+use Core\Application\Common\UseCase\NotFoundResponse;
+use Core\Application\Common\UseCase\PresenterInterface;
+use Core\Host\Application\Exception\HostException;
+use Core\Host\Application\Repository\ReadHostRepositoryInterface;
+use Core\Host\Application\Repository\WriteHostRepositoryInterface;
+use Core\Security\AccessGroup\Application\Repository\ReadAccessGroupRepositoryInterface;
+use Core\Security\AccessGroup\Domain\Model\AccessGroup;
+use Core\Service\Application\Repository\ReadServiceRepositoryInterface;
+use Core\Service\Application\Repository\WriteServiceRepositoryInterface;
+
+class DeleteHost
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private readonly ReadHostRepositoryInterface $readHostRepository,
+        private readonly WriteHostRepositoryInterface $writeHostRepository,
+        private readonly ReadServiceRepositoryInterface $readServiceRepository,
+        private readonly WriteServiceRepositoryInterface $writeServiceRepository,
+        private readonly ContactInterface $contact,
+        private readonly DataStorageEngineInterface $storageEngine,
+        private readonly ReadAccessGroupRepositoryInterface $readAccessGroupRepository,
+    ) {
+    }
+
+    /**
+     * @param int $hostId
+     * @param PresenterInterface $presenter
+     */
+    public function __invoke(int $hostId, PresenterInterface $presenter): void
+    {
+        try {
+            $this->info('Use case: DeleteHost');
+            if (! $this->contact->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_WRITE)) {
+                $this->error(
+                    "User doesn't have sufficient rights to delete a host",
+                    ['user_id' => $this->contact->getId(), 'host_id' => $hostId]
+                );
+                $presenter->setResponseStatus(
+                    new ForbiddenResponse(HostException::deleteNotAllowed())
+                );
+
+                return;
+            }
+
+            if (! $this->hostExists($hostId)) {
+                $this->error('Host not found', ['host_id' => $hostId]);
+                $presenter->setResponseStatus(new NotFoundResponse('Host'));
+
+                return;
+            }
+            $this->deleteHost($hostId);
+
```

Comment:
You need to get the host's monitoring server prior to delete and call ReadMonitoringServerRepositoryInterface->notifyConfigurationChange($monitoringServerIds)